File: taxi_sabana/check_braces.dart

import 'dart:io';

void main() {
  final file = File('lib/pantallas/autenticacion/home_conductor.dart');
  final lines = file.readAsLinesSync();
  
  int braceCount = 0;
  List<int> openLines = [];

  for (int i = 0; i < lines.length; i++) {
    String line = lines[i].trim();
    // naive comment removal
    if (line.startsWith('//')) continue;
    
    // Process char by char to handle multiple on one line
    for (int c = 0; c < line.length; c++) {
        if (line[c] == '{') {
            braceCount++;
            openLines.add(i + 1);
        } else if (line[c] == '}') {
            braceCount--;
            if (openLines.isNotEmpty) openLines.removeLast();
            
            // Print when we close a major block near our target area
            if (i > 1200 && i < 1450 && braceCount < 5) { // Assuming class + method + try is deep enough
                print('Line ${i+1}: Closing brace found. Level now: $braceCount');
            }
        }
    }
    
    if (braceCount < 0) {
        print('ERROR: Negative brace count at line ${i+1}');
        return;
    }
  }
  
  print('Final brace count: $braceCount');
  if (braceCount > 0) {
      print('Unclosed braces starting lines: ${openLines.take(5)}...');
  }
}

File: taxi_sabana/check_braces.py

def check_balance(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        lines = f.readlines()
    
    stack = []
    for i, line in enumerate(lines):
        line = line.split('//')[0] # ignore comments
        for j, char in enumerate(line):
            if char == '{':
                stack.append((i + 1, j + 1))
            elif char == '}':
                if not stack:
                    print(f"Extra closing brace at line {i + 1}, col {j + 1}")
                    return
                stack.pop()
    
    if stack:
        print(f"Unclosed braces: {len(stack)}")
        for item in stack[-1:]:
            print(f"Last unclosed brace at line {item[0]}, col {item[1]}")
    else:
        print("Braces are balanced.")

check_balance(r"e:\copia de seguridad\taxi_sabana\lib\pantallas\autenticacion\home_pasajero.dart")
